File: donarg-frontend/src/services/usuarioService.js
import api from './api'

export function registrarUsuario(data) {
    return api.post('/usuarios', data)
}

export function loginUsuario(data) {
    return api.post('/usuarios/login', data)
}

export function buscarUsuarioPorId(id) {
    return api.get(`/usuarios/${id}`)
}

export function nombreUsuarioDisponible(valor) {
    return api.get('/usuarios/nombre-usuario-disponible', { params: { valor } })
}

export function verificarEmail(token) {
    return api.get('/usuarios/verificar-email', { params: { token } })
}

export function reenviarVerificacion(id) {
    return api.post(`/usuarios/${id}/reenviar-verificacion`)
}

export function actualizarDatosPropios(data) {
    return api.patch('/usuarios/me', data)
}

export function cambiarPassword(data) {
    return api.patch('/usuarios/me/password', data)
}

export function cambiarEmail(data) {
    return api.patch('/usuarios/me/email', data)
}

export function obtenerDniPropio() {
    return api.get('/usuarios/me/dni')
}

export function subirFotoPerfil(archivo) {
    const formData = new FormData()
    formData.append('archivo', archivo)
    return api.post('/usuarios/me/foto', formData)
}

export function darDeBaja(data) {
    return api.delete('/usuarios/me', { data })
}

File: donarg-backend/src/main/java/com/donarg/api/usuario/service/UsuarioService.java
package com.donarg.api.usuario.service;

import com.donarg.api.usuario.dto.request.BajaCuentaRequest;
import com.donarg.api.usuario.dto.request.CambiarEmailRequest;
import com.donarg.api.usuario.dto.request.CambiarPasswordRequest;
import com.donarg.api.usuario.dto.request.UsuarioActualizacionRequest;
import com.donarg.api.usuario.dto.request.UsuarioLoginRequest;
import com.donarg.api.usuario.dto.request.UsuarioRegistroRequest;
import com.donarg.api.usuario.dto.response.UsuarioDniResponse;
import com.donarg.api.usuario.dto.response.UsuarioResponse;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.multipart.MultipartFile;

public interface UsuarioService {

    UsuarioResponse registrar(UsuarioRegistroRequest request, HttpServletRequest httpRequest, HttpServletResponse httpResponse);

    UsuarioResponse login(UsuarioLoginRequest request, HttpServletRequest httpRequest, HttpServletResponse httpResponse);

    void logout(HttpServletRequest httpRequest);

    UsuarioResponse verificar(Long id);

    UsuarioResponse buscarPorId(Long id);

    boolean nombreUsuarioDisponible(String nombreUsuario);

    UsuarioResponse verificarEmail(String token);

    UsuarioResponse reenviarVerificacion(Long id);

    UsuarioResponse actualizarDatosPropios(UsuarioActualizacionRequest request);

    void cambiarPassword(CambiarPasswordRequest request);

    UsuarioResponse cambiarEmail(CambiarEmailRequest request, HttpServletRequest httpRequest, HttpServletResponse httpResponse);

    UsuarioDniResponse obtenerDniPropio();

    UsuarioResponse subirFotoPerfil(MultipartFile archivo);

    void darDeBaja(BajaCuentaRequest request, HttpServletRequest httpRequest);
}
